apps/elixir_chatbot_core/lib/openai_client.ex

```diff
@@ -36,6 +36,7 @@ defmodule ElixirChatbotCore.OpenAiClient do
 
       _ when retries > 0 ->
         Logger.info("OpenAI API retry (#{retries} left)")
+        :timer.sleep(1000)
```

Comment:
Consider using Fibonnaci backoff? first retry after a milliseconds, second after b, third after c = a + b, fourth after b + c and so on - then it should not waste time on one-off faliures, while allowing time on longer fails

Reply:
True, it wasn't the best. I use a simple exponential backoff because it was the easiest to implement here